acceptance/acceptance_test.go

```diff
@@ -229,1905 +99,1417 @@ func testWithoutSpecificBuilderRequirement(
+			assert.Nil(err)
+		})
+
+		it.After(func() {
+			os.RemoveAll(tmpDir)
+		})
+
+		when("experimental is disabled", func() {
+			it("fails", func() {
+				stackManager.EnsureDefaultStackCreated()
+
+				buildpack := components.NoOpBuildpack
+				buildpackManager.PlaceBuildpacksInDir(tmpDir, buildpack)
+				builderConfigPath := builderManager.NewDynamicBuilderConfig(buildpack).ConfigFile(tmpDir)
+
+				output, err := createBuilderPack.RunWithCombinedOutput(
+					"create-builder",
+					"--no-color",
+					registry.RepoName(fmt.Sprintf("test/builder-%s", h.RandString(8))),
+					"-b", builderConfigPath,
+				)
+
+				assert.NotNil(err)
+				assert.NewOutputAssertionManager(output).ReportsWindowsContainersExperimental()
+			})
+		})
+
+		when("experimental is enabled", func() {
+			it("succeeds", func() {
+				stackManager.EnsureDefaultStackCreated()
+
+				createBuilderPack.EnableExperimental()
+
+				testBuilder := builderManager.CreateWindowsBuilder()
+				defer testBuilder.Cleanup()
+
+				assert.ImageIsWindows(testBuilder.Name())
+			})
+		})
 	})
 
-	when("stack is created", func() {
+	when("build", func() {
 		var (
-			runImageMirror string
+			assertAppImage          assertions.ImageAssertionManager
+			appName, appImageName   string
+			mockAppPath             = filepath.Join("testdata", "mock_app")
+			mixedComponentMessenger components.MixedComponents
+			cacheCleanupRequired    bool
 		)
 
 		it.Before(func() {
-			value, err := suiteManager.RunTaskOnceString("create-stack",
-				func() (string, error) {
-					runImageMirror := registryConfig.RepoName(runImage)
-					err := createStack(t, dockerCli, runImageMirror)
-					if err != nil {
-						return "", err
-					}
-
-					return runImageMirror, nil
-				})
-			h.AssertNil(t, err)
+			cacheCleanupRequired = false
+			h.SkipIf(t, containerManager.HostOS() == "windows", "These tests are not yet compatible with Windows-based containers")
 
-			suiteManager.RegisterCleanUp("remove-stack-images", func() error {
-				return h.DockerRmi(dockerCli, runImage, buildImage, value)
-			})
+			appName = fmt.Sprintf("some-org/%s", h.RandString(10))
+			appImageName = registry.RepoName(appName)
+
+			assertAppImage = assert.NewImageAssertionManager(appImageName, containerManager)
+
+			mixedComponentMessenger = components.NewMixedComponents(t, testLifecycle, pack)
+
+			cacheCleanupRequired = true
+		})
+
+		it.After(func() {
+			if cacheCleanupRequired {
+				h.DockerRmi(dockerCli, appImageName)
+				ref, err := name.ParseReference(appImageName, name.WeakValidation)
+				assert.Nil(err)
+				cacheImage := cache.NewImageCache(ref, dockerCli)
+				buildCacheVolume := cache.NewVolumeCache(ref, "build", dockerCli)
+				launchCacheVolume := cache.NewVolumeCache(ref, "launch", dockerCli)
+				cacheImage.Clear(context.TODO())
+				buildCacheVolume.Clear(context.TODO())
+				launchCacheVolume.Clear(context.TODO())
+			}
+		})
+
+		it("creates a runnable, rebuildable image on daemon from app dir", func() {
```

Comment:
This `it` is really long. What do you think about putting each "step" in a different function?

Reply:
Yes, it tests a resource intensive process that we would rather not repeat multiple times. I agree, and if this was a millisecond operation that's how I would right the tests. Unfortunately `pack build` is a multi-second operation even with the simplest of apps.

Focussing on the scope of this PR, I have intended to replicate the tests in the existing suite as closely as possible, so breaking this test up into smaller tests isn't something I'd like to include, but I'd be happy to revisit in the future.

Reply:
Sounds good.
As I wrote in another comment, since we get some good ideas from everyone, I think it's worth creating an issue and specify all of them.